sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt

```diff
@@ -1869,107 +1914,251 @@ class CompactBlockProcessor internal constructor(
-                        exception
-                    )
+                    SyncingResult.EnhanceFailed(null, exception)
                 }
             traceScope.end()
             return result
         }
 
-        // TODO [#1254]: CompactblockProcessor.fetchTransaction pass txId twice
-        // TODO [#1254]: https://github.com/zcash/zcash-android-wallet-sdk/issues/1254
+        /**
+         * Fetch the transaction complete data by [TransactionDataRequest.EnhancementRequired.txid] from Light Wallet
+         * server. This function handles [Response.Failure.Server.NotFound] by returning null.
+         *
+         * @return [RawTransactionUnsafe] if the transaction has been found, null otherwise.
+         *
+         * @throws CompactBlockProcessorException.EnhanceTransactionError in case of any other error
+         */
         @Throws(EnhanceTxDownloadError::class)
         private suspend fun fetchTransaction(
-            transactionId: String,
-            rawTransactionId: ByteArray,
-            minedHeight: BlockHeight,
-            downloader: CompactBlockDownloader
-        ): ByteArray {
+            transactionRequest: TransactionDataRequest.EnhancementRequired,
+            downloader: CompactBlockDownloader,
+        ): RawTransactionUnsafe? {
+            var transactionResult: RawTransactionUnsafe? = null
             val traceScope = TraceScope("CompactBlockProcessor.fetchTransaction")
-            var transactionDataResult: ByteArray? = null
+
             retryUpToAndThrow(TRANSACTION_FETCH_RETRIES) { failedAttempts ->
                 if (failedAttempts == 0) {
-                    Twig.debug { "Starting to fetch transaction (txid:$transactionId, block:$minedHeight)" }
+                    Twig.debug { "Starting to fetch transaction: txid: ${transactionRequest.txIdString()}" }
                 } else {
                     Twig.warn {
-                        "Retrying to fetch transaction (txid:$transactionId, block:$minedHeight) after" +
-                            " $failedAttempts failure(s)..."
+                        "Retrying to fetch transaction: txid: ${transactionRequest.txIdString()}" +
+                            " after $failedAttempts failure(s)..."
                     }
                 }
-                when (val response = downloader.fetchTransaction(rawTransactionId)) {
-                    is Response.Success -> {
-                        transactionDataResult = response.result.data
-                    }
-                    is Response.Failure -> {
-                        throw EnhanceTxDownloadError(minedHeight, response.toThrowable())
+
+                transactionResult =
+                    when (val response = downloader.fetchTransaction(transactionRequest.txid)) {
+                        is Response.Success -> response.result
+                        is Response.Failure -> {
+                            if (response is Response.Failure.Server.NotFound) {
```

Comment:
Does `downloader.fetchTransaction` convert the error strings we discovered can appear because of https://github.com/zcash/lightwalletd/issues/497 to `NotFound`? If not, this won't work properly with existing lightwalletd instances.

Reply:
This question is blocking.

Reply:
cc @str4d

Reply:
Based on this [comment](https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/pull/1512#pullrequestreview-2249143577) I think it should be handled somewhere but I wasn't able to find it in the code.

Reply:
I've created a PR for this, pleasehave a look https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/pull/1565

Reply:
The code I wrote did not handle the error string conversion, because we did not know at the time it was necessary.

Reply:
Merged #1565